packages/concerto-core/lib/decoratormanager.js

```diff
@@ -253,6 +253,251 @@ class DecoratorManager {
+                            }
+                            else {
+                                const extensionKey = dcs.name.split('Term_')[1];
+                                dictVoc[obj.declaration].propertyVocabs[obj.property][extensionKey] = dcs.arguments[0].value;
+                            }
+                        }
+                        else {
+                            if (dcs.name === 'Term'){
+                                dictVoc[obj.declaration].term = dcs.arguments[0].value;
+                            }
+                            else {
+                                const extensionKey = dcs.name.split('Term_')[1];
+                                dictVoc[obj.declaration][extensionKey] = dcs.arguments[0].value;
+                            }
+                        }
+                    }
+                });
+
+            });
+            Object.keys(dictVoc).forEach((decl)=>{
+                if (dictVoc[decl].term){
+                    strVoc += `  - ${decl}: ${dictVoc[decl].term}\n`;
+                    const otherProps = Object.keys(dictVoc[decl]).filter((str)=>str!=='term' && str!=='propertyVocabs');
+                    otherProps.forEach((key)=>{
+                        strVoc += `    ${key}: ${dictVoc[decl][key]}\n`;
+                    });
+                }
+                if (dictVoc[decl].propertyVocabs && Object.keys(dictVoc[decl].propertyVocabs).length>0){
+                    if (!dictVoc[decl].term){
+                        strVoc += `  - ${decl}: ${decl}\n`;
+                    }
+                    strVoc += '    properties:\n';
+                    Object.keys(dictVoc[decl].propertyVocabs).forEach((prop)=>{
+                        strVoc += `      - ${prop}: ${dictVoc[decl].propertyVocabs[prop].term}\n`;
+                        const otherProps = Object.keys(dictVoc[decl].propertyVocabs[prop]).filter((str)=>str!=='term');
+                        otherProps.forEach((key)=>{
+                            strVoc += `        ${key}: ${dictVoc[decl].propertyVocabs[prop][key]}\n`;
+                        });
+                    });
+                }
+            });
+            data.push(strVoc);
+        });
+        return data;
+    }
+    /**
+    * Adds a key-value pair to a dictionary (object) if the key exists,
+    * or creates a new key with the provided value.
+    *
+    * @param {Object} dictionary - The dictionary (object) to which to add the key-value pair.
+    * @param {string} key - The key to add or update.
+    * @param {any} value - The value to add or update.
+    * @param {string} declaration - The target decl.
+    * @param {string} property - The target property.
+    * @param {string} mapElement - The target mapElement.
+    * @returns {Object} - constructed DCS Dict
+    * @private
+    */
+    static constructDCSDictionary(dictionary, key, value, declaration,property,mapElement='') {
```

Comment:
I still think the last three can be part of an options object.

Reply:
Maybe you can extract out all these new methods to a separate class, like `DecoratorExtractor`, and not export it, so that it becomes an internal tool and wouldn't add bulk to the `DecoratorManager`. Then you can inject managers as a dependencies.

Reply:
Shall I create Utils instead? I can even move some of the private methods used by `decorateModels` there.

Reply:
Either way works I think. `DecoratorUtil` like we have for models, `ModelUtil` would also be a good extraction.

Reply:
Updated. Only moved the methods related to extraction for now.